erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -49,6 +49,9 @@ def set_indicator(self):
 			self.indicator_color = "green"
 			self.indicator_title = _("Paid")
 
+	def onload(self):
+		self.get("__onload").journal_entry = frappe.db.get_single_value('Accounts Settings', 'make_payment_via_journal_entry')
```

Comment:
Can you rename the flag to `make_payment_via_journal_entry `? It will be more explanatory in the client side usage. 

Try to commonify the `onload` (in accounts_controller.py?). 
And there is an existing `onload` in selling_controller.py, take care of that.